### PR TITLE
6pm 4 new database table

### PR DESCRIPTION
**Pull request by jc9907**
Created new "Search" entity and repository. Also added corresponding tests. link to issue:https://github.com/ucsb-cs156-w21/proj-mapache-search/issues/156

### CODE REVIEW

File: src/test/java/edu/ucsb/mapache/entities/SearchTests.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    assertEquals("term", s.getSearchTerm());
+  }
+
+  @Test
+  public void testSearch_getAndSetCount() {
+    Search s = new Search();
+    s.setCount(123);
+    assertEquals(123, s.getCount());
+  }
+}
```
Comment (pconrad) [inline src/test/java/edu/ucsb/mapache/entities/SearchTests.java:28]: Once you add the .equals, .toString, and .hashCode, you can consult this file for a good example of how to test a class like this one:

https://github.com/ucsb-cs156-w21/proj-ucsb-cs-las/blob/main/src/test/java/edu/ucsb/ucsbcslas/entities/TutorAssignmentTests.java

Note that you can test all of the getters and setters with one line of code:

```
@Test
  public void testGettersAndSetters() throws Exception {
    // See: https://github.com/codebox/javabean-tester
    JavaBeanTester.test(TutorAssignment.class);
  }
```

**Review by pconrad — CHANGES_REQUESTED** [review]
I have a couple of suggestions.   These will make the code a lot easier to maintain down the road.

File: src/main/java/edu/ucsb/mapache/entities/Search.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  private int count;
+
+  public Search(){
+
+  }
+
+  public Search(Long id, String searchTerm, int count){
+      this.id = id;
+      this.searchTerm = searchTerm;
+      this.count = count
```
Comment (pconrad) [inline src/main/java/edu/ucsb/mapache/entities/Search.java:26]:
```
 Error:  /home/runner/work/proj-mapache-search/proj-mapache-search/src/main/java/edu/ucsb/mapache/entities/Search.java:[26,25] ';' expected`
```

Need to fix this syntax error.

**Review by pconrad — CHANGES_REQUESTED** [review]

**Review by chengyanwu — APPROVED** [review]
LGTM

**Comment by jayleenli** [comment]
Approving although not my group. Jiachen went to my office hours and also discussed the codecov issues after lecture on Thursday with me, we finally got it working after office hours. They have also been blocked on this issue for a couple days and other team members cannot make progress unless this one is complete

**Review by jayleenli — APPROVED** [review]

**Pull request merged**